packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -696,9 +696,8 @@ export class AngularCompilerPlugin {
               const dependencies = Object.keys(this._lazyRoutes)
                 .map((key) => {
                   const modulePath = this._lazyRoutes[key];
-                  const importPath = key.split('#')[0];
                   if (modulePath !== null) {
-                    const name = importPath.replace(/(\.ngfactory)?(\.(js|ts))?$/, '');
+                    const name = key.split('#')[0];
```

Comment:
`const name = importPath.replace(/(\.ngfactory)?\.(js|ts)$/, '');` was originally there since 6.2.0

Reply:
Do you know why it was there? Pretty much this RegExp, is not triggered and if it did. It will break lazy loading as `ngfactory` cannot be replaced with a ''.

Reply:
probably not.  Let's get rid of it.